src/compare.py

```diff
@@ -126,6 +133,59 @@ def compare_phash(source: MatLike, capture: MatLike, mask: MatLike | None = None
     return 1 - (hash_diff / 64.0)
 
 
+# copied from https://github.com/madmaze/pytesseract
+def subprocess_args():
+    # See https://github.com/pyinstaller/pyinstaller/wiki/Recipe-subprocess
+    # for reference and comments.
+
+    kwargs = {
+        "stdin": subprocess.PIPE,
+        "stdout": subprocess.PIPE,
+        "stderr": subprocess.DEVNULL,
+        "startupinfo": None,
+        "env": environ,
+    }
+
+    if hasattr(subprocess, "STARTUPINFO"):
+        kwargs["startupinfo"] = subprocess.STARTUPINFO()
+        kwargs["startupinfo"].dwFlags |= subprocess.STARTF_USESHOWWINDOW
+        kwargs["startupinfo"].wShowWindow = subprocess.SW_HIDE
+
+    return kwargs
+
+
+def run_tesseract(capture: MatLike):
+    png = np.array(cv2.imencode(".png", capture)[1]).tobytes()
+    p = subprocess.Popen(TESSERACT_CMD, **subprocess_args())
+    output = p.communicate(input=png)[0]
+    return output.decode(DEFAULT_ENCODING)
+
+
+def extract_and_compare_text(capture: MatLike, texts: list[str]):
+    """
+    Compares the extracted text of the given image and returns the similarity between the two texts.
+    The best match of all texts is returned.
+
+    @param capture: Image of any given shape as a numpy array
+    @param texts: a list of strings to match for
+    @return: The similarity between the text in the image and the text supplied as a number 0 to 1.
+    """
+    # if the string is found 1:1 in the string extracted from the image a 1 is returned.
+    # otherwise the levenshtein ratio is calculated between the two strings and gets returned.
+    image_string = run_tesseract(capture).lower().strip()
```

Comment:
Is there an issue with supporting casing? `A` =/= `a` when you already know what text to expect.

Reply:
I went with lowercase and truncation to increase the matching accuracy as the OCR is not perfect at recognizing upper and lowercase in the many different game fonts and renderings.

Reply:
Thinking a bit more over it, here's a different wording that makes sense to me:

```suggestion
    # Especially with stylised characters, OCR could conceivably get the right letter, but mix up the casing (m/M, o/O, t/T, etc.)
    image_string = run_tesseract(capture).lower().strip()
```